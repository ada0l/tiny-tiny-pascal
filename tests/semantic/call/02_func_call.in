function f(i: Integer; c: Double; c_cast: Double; b: Boolean; s: string): integer;
begin f := 10; end;

begin
f(100, 20.3, 10, True, 'hello, world');
end.