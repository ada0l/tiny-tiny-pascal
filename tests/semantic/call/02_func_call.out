main
   program name:
      without name
   decls:
      function decl
         f 
         type:
            integer 
         param
            without modifier
            i 
            type:
               Integer 
         param
            without modifier
            c 
            type:
               Double 
         param
            without modifier
            c_cast 
            type:
               Double 
         param
            without modifier
            b 
            type:
               Boolean 
         param
            without modifier
            s 
            type:
               string
         decls:
         block:
            compounds statements
               :=
                  f 
                  10
   block:
      compounds statements
         f 
            100
            20.3
            cast
               10
            True
            'hello, world'
------------------------------------------------------------------------------------------------------------------------------------------------------
| id                          | type of object              | ret type                                                                               |
------------------------------------------------------------------------------------------------------------------------------------------------------
| boolean                     | type                        | boolean                                                                                |
| integer                     | type                        | integer                                                                                |
| double                      | type                        | double                                                                                 |
| char                        | type                        | char                                                                                   |
| string                      | type                        | string                                                                                 |
| writeln                     | procedure                   | writeln                                                                                |
| write                       | procedure                   | write                                                                                  |
| read                        | procedure                   | read                                                                                   |
| random                      | function                    | double                                                                                 |
======================================================================================================================================================
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| sqrt                        | function                    | double                                                                                 |
======================================================================================================================================================
| x                           | param                       | double                                                                                 |
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------------------------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------------------------------------------
| id                          | type of object              | ret type                                                                               |
------------------------------------------------------------------------------------------------------------------------------------------------------
| f                           | function                    | integer                                                                                |
======================================================================================================================================================
| i                           | param                       | integer                                                                                |
| c                           | param                       | double                                                                                 |
| c_cast                      | param                       | double                                                                                 |
| b                           | param                       | boolean                                                                                |
| s                           | param                       | string                                                                                 |
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------------------------------------------------------------------------------------------------------------------

